src/Service/RevisionSaver.php

```diff
@@ -36,6 +37,14 @@ public function __construct( MediawikiApi $api ) {
 	 * @returns bool success
 	 */
 	public function save( Revision $revision, EditInfo $editInfo = null ) {
+		if( is_null( $editInfo ) ) {
+			if( is_null( $revision->getEditInfo() )) {
```

Comment:
Right, there is always an editInfo in revision (even if it can be zero) and so the second check isn't necessary.